src/server/index.js

```diff
@@ -183,45 +183,45 @@ export default async function startServer(options) {
     )
   );
 
-  router.post("/api/files/:filePath*", koaBody({ multipart: true }), async ctx => {
+  router.post("/api/files/:filePath*", async ctx => {
     const filePath = ctx.params.filePath ? path.resolve(projectPath, ctx.params.filePath) : projectPath;
 
     if (ctx.request.query.open) {
+      // Attempt to open file at filePath with the default application for that file type.
       opn(filePath);
+    } else if (ctx.request.query.mkdir) {
+      // Make the directory at filePath if it doesn't already exist.
+      await fs.ensureDir(filePath);
+    } else {
+      // If uploading as text body, write it to filePath using the stream API.
+      const writeStream = fs.createWriteStream(filePath, { flags: "w" });
 
-      ctx.body = {
-        success: true
-      };
-    } else if (ctx.request.files && ctx.request.files.file) {
-      const file = ctx.request.files.file;
+      ctx.req.pipe(writeStream);
 
-      await fs.rename(file.path, filePath);
+      await new Promise((resolve, reject) => {
+        function cleanUp() {
+          // eslint-disable-next-line
+          writeStream.removeListener("finish", onFinish);
+          // eslint-disable-next-line
```

Comment:
Please use the strict form of this disable comment: `// eslint-disable-next-line no-use-before-define`
Or just configure the rule to allow for use of functions before they're defined.